src/shared/ui/common/location.ts

```diff
@@ -0,0 +1,97 @@
+/*!
+ * Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import * as vscode from 'vscode'
+import * as os from 'os'
+import { addCodiconToString } from '../../utilities/textUtilities'
+import * as nls from 'vscode-nls'
+import { WizardControl, WIZARD_RETRY } from '../../wizards/wizard'
+import { createQuickPick, DataQuickPickItem, QuickPickPrompter } from '../pickerPrompter'
+import { PrompterButtons } from '../buttons'
+
+const localize = nls.loadMessageBundle()
+
+interface Folder {
+    readonly uri: vscode.Uri
+    readonly name: string
+}
+
+class FolderQuickPickItem implements Omit<DataQuickPickItem<vscode.Uri>, 'disabled'> {
```

Comment:
> This seems way overkill for a class. Just use an interface or even an object directly (why not combine with above Folder interface?)

Yeah I know. I just modified it in-place though it should be refactored further.

> ??? re: omission? Doesn't look like disabled is on the type. Same below.

This is because its a class trying to implement a type where `disabled` can't be inferred. 
Will disappear when the class goes away.